apps/api-server/src/adapter/openstad/router.js

```diff
@@ -294,13 +294,24 @@ router
     }
     return next();
   })
-  .get(function (req, res, next) {
+  .get(async function (req, res, next) {
+    // Check if redirectUri is in allowlist
+    if(req.query.redirectUri){
+      // Get project by req.params.projectId
+      const projectId = req.params.projectId;
+      const project = await db.Project.findByPk(projectId);
+      if(!project) return next(createError(404, 'Project not found'));
+
+      // Check if redirectUri is in allowlist
+      let allowedDomains = project?.config?.widgets?.allowedDomains || [];
+      if(allowedDomains.includes(req.query.redirectUri)){
```

Comment:
De `allowedDomains` bevat een lijst met domeinen (bijv. `openstad.org`), en geen volledige paden. Je vergelijkt nu met de `req.query.redirectUri` die wel een protocol en een path kan bevatten (bijv. `https://openstad.org/plannen?resourceId=4`). Dan gaat deze `includes` check niet werken.

Het is beter om een `new URL(...)` te gebruiken, en vanuit het object wat je terugkrijgt de `host` te vergelijken met de `allowedDomains`:

![image](https://github.com/openstad/openstad-headless/assets/205926/cea2170c-8033-4840-9eeb-8a48a4387644)

Reply:
Nogal een oversight inderdaad, heb 't aangepast.